sonar-db/src/main/resources/org/sonar/db/measure/MeasureMapper.xml

```diff
@@ -29,41 +29,55 @@
     select
     <include refid="measureColumns"/>
     from
-      project_measures pm
-      inner join projects p on p.uuid=pm.component_uuid
-      inner join snapshots analysis on analysis.component_uuid = p.project_uuid and analysis.uuid = pm.analysis_uuid
-      <if test="query.getMetricKeys() != null">
-        inner join metrics m on m.id = pm.metric_id
-      </if>
+    project_measures pm
```

Comment:
Indeed you can say there is no duplication, but this query will be a nightmare to maintain. It's hard to tell which SQL request can be generated in the end, therefor hard to tell whether the request is correct or not.
(I tried, it is hard to read and understand)

From a UT point of view, having duplications is the same as this monster request but it is way more maintainable and duplications can be reduce by the mean of SQL includes

Some conditions on the Query object can't be true at the same time but this is not visible in the XML (which means the request looks more complexe that it actually is). This might be the way to follow to know which are the various SQL requests to write in the XML.